cdap-ui/app/cdap/components/AbstractWidget/DLPCustomWidget/DLPRow.tsx

```diff
@@ -0,0 +1,398 @@
+import ConfigurationGroup from 'components/ConfigurationGroup';
+import { IErrorObj } from 'components/ConfigurationGroup/utilities';
+import {
+  extractAndSplitMatchingErrors,
+  parseNestedErrors,
+  parseTransformOptions,
+} from './utilities/index';
+
+const styles = (theme): StyleRules => {
+  return {
+    ...AbstractRowStyles(theme),
+    root: {
+      minHeight: '50px',
+      display: 'grid',
+      gridTemplateColumns: '1fr auto auto auto',
+      alignItems: 'center',
+      '& > button': {
+        width: '40px',
+        height: '40px',
+        margin: 'auto',
+        marginTop: '5px',
+      },
+    },
+    inputContainer: {
+      display: 'grid',
+      gridTemplateColumns: '40px 1fr 30px 1fr 50px 1fr',
+      gridGap: '10px',
+    },
+    errorBorder: {
+      border: '2px solid',
+      borderColor: theme.palette.red[50],
+    },
+    disabled: {
+      color: `${theme.palette.grey['50']}`,
+    },
+    separator: {
+      textAlign: 'center',
+    },
+    group: {
+      marginLeft: '20px',
+      marginTop: '15px',
+      marginBottom: '15px',
+      '& > div > div': {
+        padding: '15px 10px 10px',
+      },
+    },
+    groupTitle: {
+      marginBottom: '15px',
+      '& > h2': {
+        fontSize: '1.2rem',
+      },
+    },
+    errorText: {
+      color: theme.palette.red[50],
+      marginLeft: '15px',
+      marginTop: '5px',
+    },
+  };
+};
+
+export interface IErrorConfig {
+  transform: string;
+  fields: string;
+  filters: string;
+  transformPropertyId: string;
+  isNestedError: boolean;
+}
+
+export interface ITransformProp {
+  label: string;
+  name: string;
+  options: IWidgetProperty[];
+  supportedTypes?: string[];
+  filters?: IPropertyFilter[];
+}
+export type FilterOption = IOption | string;
+
+export interface IDLPRowProps extends IAbstractRowProps<typeof styles> {
+  transforms: ITransformProp[];
+  filters: FilterOption[];
+  extraConfig: any;
+}
+
+export interface IDLPRowState {
+  fields: string;
+  transform: string;
+  filters: string;
+  transformProperties: Record<string, string>;
+  expanded?: boolean;
+  transformObj?: ITransformProp;
+}
+
+type StateKeys = keyof IDLPRowState;
+class DLPRow extends AbstractRow<IDLPRowProps, IDLPRowState> {
+  public static defaultProps = {
+    transforms: [],
+    filters: [],
+    extraConfig: '',
+    expanded: false,
+  };
+
+  public state = {
+    fields: '',
+    transform: '',
+    filters: '',
+    transformProperties: {},
+    expanded: false,
+    transformObj: null,
+  };
+
+  public componentDidMount() {
+    try {
+      const jsonString = this.props.value;
+      const jsonObj = JSON.parse(jsonString);
+      jsonObj.transformObj =
+        this.props.transforms.filter((t) => t.name === jsonObj.transform)[0] || null;
+      this.configurationGroupPluginProperties = parseTransformOptions(jsonObj.transformObj);
+      this.configurationGroupWidgetJson =
+        jsonObj.transformObj !== null
```

Comment:
nit: here as well. Please switch it to `if...else` condition

Reply:
done